lambda/simplif.ml

```diff
@@ -25,7 +25,7 @@ open Debuginfo.Scoped_location
 exception Real_reference
 
 let rec eliminate_ref id = function
-    Lvar v as lam ->
+    Lvar v | Lmutvar v as lam ->
       if Ident.same v id then raise Real_reference else lam
```

Comment:
I believe that `eliminate_ref` is only called with pure variables `id`, so `Ident.same v id` is impossible in the `Lmutvar` case. Could you simplify the code with `Lmutvar _ as lam -> lam` directly?